### PR TITLE
fix(definitions): feed name incorrectly small

### LINKED ISSUE #16: Group feeds not working?
Board : ESP8266 Feather Huzzah 12S
Sketches : https://gist.github.com/j15e/2a91c09f8e4a0a12a187e32fee770615

The `group.c` is not receiving any message whereas `single-feed.c` works as expected. I also have try to set / publish a value and it doesn't work via the group class either. 

I do have the feed `pump` setup in the `garden` group (I think?) ;

capture d ecran 2017-09-27 a 22 51 06

### LINKED ISSUE #119: Incorrect handling of data->feedName() when feed has group
Thank you for providing this excellent data warehouse and associated apis!

I found that when using a feed scoped to a group that's longer than the [20 character limit](https://github.com/adafruit/Adafruit_IO_Arduino/blob/ce83ba999056639728de655fbad49bfb4c935cfd/src/AdafruitIO_Definitions.h#L127), the `feedName()` function call overflows incorrectly into the data.

- Arduino board:  NodeMCU 1.0 ESP-12E Module (ESP 8266)

- Arduino IDE version (found in Arduino -> About Arduino menu):  1.8.12

- Steps to reproduce the problem below:

1) Create a feed group, eg: `coop-command`.
2) Create a feed that will cause the group name + . + feed to be > 20; eg: `leds_on_off`
3) Create a dashboard with an on/off switch whose texts are "ON" and "OFF"
4) Create and run a sketch with the following code:

```c++
#include <AdafruitIO_WiFi.h>

bool ledsOn = false;
AdafruitIO_WiFi io(ENV_AIO_USERNAME, ENV_AIO_KEY, ENV_WIFI_SSID, ENV_WIFI_PASS);
AdafruitIO_Feed *ledOnOffFeed = io.feed("coop-command.leds_on_off");
void handleAdafruitMessage(AdafruitIO_Data *data)
{
  String dataFeed = data->feedName();
  String dataString = data->toString();
  Serial.println("[adafruit] " + String(data->feedName()) + " <- " + dataString);

 if (dataFeed == "coop-command.leds_on_off" && (dataString == "ON" || dataString == "OFF"))
  {
    if (dataString == "ON")
    {
      ledsOn = true;
    }
    else
    {
      ledsOn = false;
    }

    return;
  }

  Serial.println("[adafruit] malformed message, ignored.");
}
void connectAdafruitIO()
{
  // connect to io.adafruit.com
  Serial.print("[adafruit] connecting");
  io.connect();

  // set up a message handlers for the feeds
  ledOnOffFeed->onMessage(handleAdafruitMessage);

  // wait for a connection
  while (io.status() < AIO_CONNECTED)
  {
    Serial.print(".");
    delay(500);
  }

  // we are connected
  Serial.println();
  Serial.println(io.statusText());

  // get initial states
  ledOnOffFeed->get();
}

void serialInit()
{
  // set up serial monitor and wait for it to open
  Serial.begin(9600);

  do
  {
    delay(100);
  } while (!Serial);

  Serial.println();
}

void setup()
{
  serialInit();
  connectAdafruitIO();
}

void loop() {
  io.run();
}
```

5) toggle the on / off switch and observe the overflow in the serial monitor:

```
[adafruit] coop-command.leds_onON,,,
 <- ON
[adafruit] malformed message, ignored.
```

### PR DESCRIPTION
**Pull request by blaskovicz**
Grouped feeds can be a max of 258 characters long (128 for the group, a single dot,
and another 128 for the feed name).

Fixes #119

### CODE REVIEW

**Comment by blaskovicz** [comment]
will look at it when i have time, thanks for the feedback 👍

File: src/AdafruitIO_Definitions.h [hunk 1]
```diff
@@ -124,11 +124,15 @@ class AdafruitIOGroupCallback {
...
 
-#define AIO_FEED_NAME_LENGTH 20 ///< Maximum length of an Adafruit IO Feed name
+#define AIO_FEED_NAME_LENGTH                                                   \
+  258 ///< Maximum length of an Adafruit IO Feed    \
+                                  ///< Name; 128 + 1 + 128 for the group, a dot \
+                                  ///< , and actual feed name.
 #define AIO_DATA_LENGTH                                                        \
   45 ///< Maximum length of data sent/recieved from Adafruit IO
 #define AIO_CSV_LENGTH                                                         \
-  150 ///< Maximum comma-separated-value length from Adafruit IO
```
Comment (brentru) [inline src/AdafruitIO_Definitions.h:131]: Why `+4`?

Comment (blaskovicz) [reply src/AdafruitIO_Definitions.h:131]: https://github.com/adafruit/Adafruit_IO_Arduino/issues/127#issuecomment-654242578

Comment (brentru) [reply src/AdafruitIO_Definitions.h:131]: @blaskovicz - ok, rest looks good. I'll test by making some long feed names and testing on physical hardware tomorrow.

Comment (blaskovicz) [reply src/AdafruitIO_Definitions.h:131]: @brentru any luck?

File: library.properties [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name=Adafruit IO Arduino
-version=3.7.0
+version=3.7.1
```
Comment (brentru) [inline library.properties:2]: this needs to be bumped , I'll bump from my end

**Comment by brentru** [comment]
This tests properly on hardware.

To test,  I created a grouped feed with len 128 and then used example `adafruitio_11_group_pub`:
```
Adafruit IO connected.
sending example.ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff-> 0
sending example.count-2 -> 0
```
On io web, `example.ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff` correctly obtained value 0

**Pull request merged**